ethereum/core/src/main/java/org/hyperledger/besu/ethereum/trie/diffbased/bonsai/BonsaiAccount.java

```diff
@@ -173,6 +173,22 @@ public String toString() {
         + '}';
   }
 
+  @Override
+  public boolean equals(final Object o) {
```

Comment:
We should move as much of this up into DiffBasedAccount as possible so that other implementations inherit this (like Verkle).  perhaps implement equals in DiffBasedAccount and delegate to an abstract method that implementations are forced to provide.